### PR TITLE
Decimal remainder, pow, div

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso [hunk 1]
```diff
@@ -519,6 +518,127 @@ type Decimal
...
+    ## ALIAS modulo, modulus
+       GROUP Operators
+       ICON math
+       Computes the remainder when dividing this by that.
+
+       Arguments:
+       - that: The number to divide this by.
+
+       Remainder in Enso will undergo automatic conversions such that you need
+       not convert other numeric types to `Decimal` manually.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso]: This sounds to me like the _result_ of the operation will undergo conversions, which as far as I understand is not the case and you mean the _arguments_, right?

Maybe we can rephrase it to make it clearer?
```suggestion
       Arguments of the `remainder` operation will undergo automatic conversions such that you need
       not convert other numeric types to `Decimal` manually.
```

maybe something like that?
1. I don't think you need to say `in Enso` there - this is an Enso library so that is rather assumed,
2. IMO it's good to point you mean _arguments_

File: test/Base_Tests/src/Data/Decimal_Spec.enso [hunk 2]
```diff
@@ -492,6 +492,74 @@ add_specs suite_builder =
...
+                + [[5, 3, 2], [5.0, 3.0, 2.0], [3.5, 2, 1.5], [10.5, 1.0, 0.5], [3, 1, 0], [3.0, 1.0, 0]]
+                + [[-5, 3, -2], [-5.0, 3.0, -2.0], [-3.5, 2, -1.5], [-10.5, 1.0, -0.5], [-3, 1, 0], [-3.0, 1.0, 0]]
+                + [["9223372036854775807", 10, 7], ["9223372036854775808", 10, 8], ["922337203685477580000000000008", 10, 8]]
+                + [["-9223372036854775808", 10, -8], ["-9223372036854775809", 10, -9], ["-922337203685477580000000000008", 10, -8]]
+            cases.map c->
+                base = c.at 0
+                modulus = c.at 1
+                residue = c.at 2
+                (Decimal.new base % modulus) . should_equal residue
+                (Decimal.new base % Decimal.new modulus) . should_equal residue
```
Comment (radeusgd) [inline test/Base_Tests/src/Data/Decimal_Spec.enso]: I'd probably add a test case for something like `2.7 % 0.5`. Should it work and return `0.2` or are non-integer arguments not allowed? Either way it would be good to have a test.

File: test/Base_Tests/src/Data/Numbers_Spec.enso [hunk 3]
```diff
@@ -101,6 +101,9 @@ add_specs suite_builder =
             (3 / 4) . should_equal 0.75 epsilon=eps
             (almost_max_long * 2 / almost_max_long_times_three) . should_equal 0.6666666 epsilon=eps
 
+            1/0 . is_infinite . should_be_true
+            -1/0 . is_infinite . should_be_true
```
Comment (radeusgd) [inline test/Base_Tests/src/Data/Numbers_Spec.enso:105]: not a problem, but out of curiosity - why is this test moved?

**Review by radeusgd — APPROVED** [review]
Looks good, just some small comments

**Review by AdRiley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso (hunk 1 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso]: Done.

File: test/Base_Tests/src/Data/Decimal_Spec.enso (hunk 2 above)
Comment (GregoryTravis) [reply test/Base_Tests/src/Data/Decimal_Spec.enso]: Done.

File: test/Base_Tests/src/Data/Numbers_Spec.enso (hunk 3 above)
Comment (GregoryTravis) [reply test/Base_Tests/src/Data/Numbers_Spec.enso:105]: Because it was originally with the modulo tests, and I thought it should be with the floating point division tests.

Comment (radeusgd) [reply test/Base_Tests/src/Data/Numbers_Spec.enso:105]: Ah, makes sense
